luaui/Widgets/cmd_onoff.lua

```diff
@@ -0,0 +1,39 @@
+function widget:GetInfo()
+	return {
+		name = "Onoff for Hound and trajectory",
+		desc = "onoff action will now also switch between Hound weapons and high/low trajectory",
+		author = "Lexon",
+		date = "19.08.2022",
+		license = "GNU GPL, v2 or later",
+		layer = 0,
+		enabled = true, --enabled by default
+	}
+end
+
+local spGetSelectedUnits = Spring.GetSelectedUnits
+local spGetUnitStates = Spring.GetUnitStates
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+
+local function onoff()
+	local selectedUnits = spGetSelectedUnits()
+
+	if selectedUnits[1] == nil then return end
+
+	local weapon = spGetUnitStates(selectedUnits[1])["active"]
+
+	if weapon == true then
+		weapon = 0
+	elseif weapon == false then
+		weapon = 1
+	else
+		return
+	end
+
+	for _, unit in pairs(selectedUnits) do
+		spGiveOrderToUnit(unit, CMD.ONOFF, { weapon }, 0)
+	end
```

Comment:
Does this have performance issues if a lot of units are selected? If 1,000 units are selected, but the On/Off command only applies to one, it is still trying to issue the order to the other 999.

Reply:
It's possible this is more performant than checking, as it avoids invoking `GetUnitStates` for each unit, delegating the check to the engine (or some callin). Only profiling can tell

Reply:
Issuing an order to a unit invokes additional callins, particularly `CommandFallback` and `AllowCommand`, so it is probably better to prune beforehand